src/Test/L0/Worker/Release/JenkinsArtifactL0.cs

```diff
@@ -187,48 +187,48 @@ public sealed class JenkinsArtifactL0
-        //             _ec.Verify(x => x.QueueAttachFile(It.Is<string>(y => y.Equals(CoreAttachmentType.FileAttachment)), It.IsAny<string>(), It.Is<string>(z => string.Join("", File.ReadAllLines(z)).Contains(expectedCommitUrl))), Times.Once);
-        //         }
-        //         finally
-        //         {
-        //             IOUtil.DeleteDirectory(commitRootDirectory, CancellationToken.None);
-        //         }
-        //     }
-        // }
+        [Fact]
+        [TraitAttribute("Level", "L0")]
+        [TraitAttribute("Category", "Worker")]
+        public async void CommitsShoulHaveUrlIfItsGitRepo()
+        {
+            using (TestHostContext tc = Setup())
+            {
+                string commitRootDirectory = tc.GetDirectory(WellKnownDirectory.Root);
+
+                try
+                {
+                    JenkinsArtifactDetails details = _artifactDefinition.Details as JenkinsArtifactDetails;
+                    details.StartCommitArtifactVersion = "10";
+                    details.EndCommitArtifactVersion = "20";
+
+                    var artifact = new JenkinsArtifact();
+                    artifact.Initialize(tc);
+
+                    SetupBuildRangeQuery(details, "{ \"allBuilds\": [{ \"number\": 20 }, { \"number\": 10 }, { \"number\": 2 } ] }");
+                    string commitResult = " {\"builds\": [{ \"number\":9, \"result\":\"SUCCESS\", \"changeSet\": { \"items\": [{ \"commitId\" : \"2869c7ccd0b1b649ba6765e89ee5ff36ef6d4805\", \"author\": { \"fullName\" : \"testuser\" }, \"msg\":\"test\" }]}}]}";
+                    string commitsUrl = $"{details.Url}/job/{details.JobName}/api/json?tree=builds[number,result,changeSet[items[commitId,date,msg,author[fullName]]]]{{0,1}}";
+                    _httpClient.Setup(x => x.GetStringAsync(It.Is<string>(y => y.StartsWith(commitsUrl)), It.IsAny<string>(), It.IsAny<string>(), It.IsAny<bool>()))
+                                .Returns(Task.FromResult(commitResult));
+
+                    string repoUrl = $"{details.Url}/job/{details.JobName}/{details.EndCommitArtifactVersion}/api/json?tree=actions[remoteUrls],changeSet[kind]";
+                    string repoResult = "{ \"actions\": [ { \"remoteUrls\": [ \"https://github.com/TestUser/TestRepo\" ] }, ], \"changeSet\": { \"kind\": \"git\" } }";
+                    _httpClient.Setup(x => x.GetStringAsync(It.Is<string>(y => y.StartsWith(repoUrl)), It.IsAny<string>(), It.IsAny<string>(), It.IsAny<bool>()))
+                                .Returns(Task.FromResult(repoResult));
+
+                    string commitFilePath = Path.Combine(commitRootDirectory, $"commits_{details.Alias}_1.json");
+                    Directory.CreateDirectory(commitRootDirectory);
+
+                    string expectedCommitUrl = "https://github.com/TestUser/TestRepo/commit/2869c7ccd0b1b649ba6765e89ee5ff36ef6d4805";
+                    await artifact.DownloadCommitsAsync(_ec.Object, _artifactDefinition, commitRootDirectory);
```

Comment:
I had to change the URL from commits to commit, as commits will be landing in the commits list page. Rather it should land in the specific commit's page. Noticed this and change the code later, but failed to fix the UT.